File: app/src/main/java/com/blackbelt/androidboundrv/view/AndroidMVVMActivity.java
package com.blackbelt.androidboundrv.view;

import com.blackbelt.androidboundrv.App;
import com.blackbelt.androidboundrv.R;
import com.blackbelt.androidboundrv.manager.MoviesManager;
import com.blackbelt.androidboundrv.view.adapter.FragmentModel;
import com.blackbelt.androidboundrv.view.adapter.MainPagerAdapter;
import com.blackbelt.androidboundrv.view.androidmvvm.AndroidMoviesFragment;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

public class AndroidMVVMActivity extends AppCompatActivity {

    Unbinder mUnbinder;

    @BindView(R.id.main_activity_viewpager)
    ViewPager mViewPager;

    @BindView(R.id.main_activity_tabs)
    TabLayout mTabLayout;

    @BindView(R.id.toolbar)
    Toolbar mToolbar;

    @Inject
    MoviesManager mMoviesManager;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mUnbinder = ButterKnife.bind(this);
        App.getComponent().inject(this);

        setupTabs();
        setSupportActionBar(mToolbar);
    }


    private void setupTabs() {

        List<FragmentModel> fragmentModel = new ArrayList<>();

        Bundle bundle = new Bundle();
        bundle.putBoolean("movies", true);

        fragmentModel.add(new FragmentModel(getResources().getString(R.string.movies),
                AndroidMoviesFragment.class, bundle));

        bundle = new Bundle();
        bundle.putBoolean("movies", false);

        fragmentModel.add(new FragmentModel(getResources().getString(R.string.tvshows),
                AndroidMoviesFragment.class, bundle));
        mViewPager.setAdapter(new MainPagerAdapter(this, fragmentModel));
        mTabLayout.setupWithViewPager(mViewPager);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mUnbinder != null) {
            mUnbinder.unbind();
        }
    }
}

File: app/src/main/java/com/blackbelt/androidboundrv/view/AndroidMVVMActivity.kt
package com.blackbelt.androidboundrv.view

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import com.blackbelt.androidboundrv.R
import com.blackbelt.androidboundrv.R.id.*
import com.blackbelt.androidboundrv.view.adapter.FragmentModel
import com.blackbelt.androidboundrv.view.adapter.MainPagerAdapter
import com.blackbelt.androidboundrv.view.androidmvvm.AndroidMoviesFragment
import dagger.android.AndroidInjection
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class AndroidMVVMActivity : BaseActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupTabs()
        setSupportActionBar(findViewById(R.id.toolbar))
    }

    private fun setupTabs() {

        val fragmentModel = ArrayList<FragmentModel>()

        var bundle = Bundle()
        bundle.putBoolean("movies", true)

        fragmentModel.add(FragmentModel(resources.getString(R.string.movies),
                AndroidMoviesFragment::class.java, bundle))

        bundle = Bundle()
        bundle.putBoolean("movies", false)

        fragmentModel.add(FragmentModel(resources.getString(R.string.tvshows),
                AndroidMoviesFragment::class.java, bundle))
        val viewPager = findViewById<ViewPager>(R.id.main_activity_viewpager)
        val tabs = findViewById<TabLayout>(R.id.main_activity_tabs)
        viewPager.adapter = MainPagerAdapter(this, fragmentModel)
        tabs.setupWithViewPager(viewPager)

    }

}
